test/query_test.js

```diff
@@ -1,452 +1,564 @@
-import {assert} from 'chai'
-import {BadRequest, NotFound} from '../src/errors'
-import {FaunaDate, FaunaTime, Ref, SetRef} from '../src/objects'
-import * as query from '../src/query'
-import {assertRejected, client, getClient} from './util'
-
-let classRef, nIndexRef, mIndexRef, refN1, refM1, refN1M1, thimbleClassRef
-
-describe('query', () => {
-  before(async () => {
-    classRef = (await client.post('classes', {name: 'widgets'})).ref
-
-    nIndexRef = (await client.post('indexes', {
-      name: 'widgets_by_n',
-      source: classRef,
-      path: 'data.n',
-      active: true
-    })).ref
-
-    mIndexRef = (await client.post('indexes', {
-      name: 'widgets_by_m',
-      source: classRef,
-      path: 'data.m',
-      active: true
-    })).ref
-
-    refN1 = (await create({n: 1})).ref
-    refM1 = (await create({m: 1})).ref
-    refN1M1 = (await create({n: 1, m: 1})).ref
-
-    thimbleClassRef = (await client.post('classes', {name: 'thimbles'})).ref
-  })
+'use strict';
+
+var assert = require('chai').assert;
+var errors = require('../src/errors');
+var objects = require('../src/objects');
+var query = require('../src/query');
+var util = require('./util');
+var Promise = require('es6-promise').Promise;
+
+var FaunaDate = objects.FaunaDate,
+  FaunaTime = objects.FaunaTime,
+  Ref = objects.Ref,
```

Comment:
Ref is quite a common objects that needs to be built. I think all relevant Values should have corespondent functions on the `query` namespace so the user doesn't have to import them all.